src/class-wpml-gutenberg-integration.php

```diff
@@ -257,17 +257,30 @@ private function guess_inner_HTML_parts( $block ) {
 
 		$parts = array( $inner_HTML, '' );
 
-		preg_match( '#>\s*</#', $inner_HTML, $matches );
-
-		if ( count( $matches ) === 1 ) {
-			$parts = explode( $matches[0], $inner_HTML );
-			if ( count( $parts ) === 2 ) {
-				$match_mid_point = 1 + ( strlen( $matches[0] ) - 3 ) / 2;
-				// This is the first ">" char plus half the remaining between the tags
-
-				$parts[0]        .= substr( $matches[0], 0, $match_mid_point );
-				$parts[1]        = substr( $matches[0], $match_mid_point ) . $parts[1];
-			}
+		switch( $block->blockName ) {
+			case 'core/media-text':
+				$html_to_find = '<div class="wp-block-media-text__content">';
+				$pos          = mb_strpos( $inner_HTML, $html_to_find ) + mb_strlen( $html_to_find );
+				$parts        = array(
+					substr( $inner_HTML, 0, $pos ),
```

Comment:
@adrianosferreira, you can use `mb_substr` here and below.

Reply:
done.